### DISCUSSION TITLE
I need some guidance on how to use this library in a custom way

### DISCUSSION THREAD

**Discussion by patchthecode** [appium/java-client#1754] [Q&A]
It's my understanding that this library converts scripts into JsonWireprotocol format, before sending to the server.
Can someone guide me to the location in the code where the script is converted into this format?

**Comment by KazuCocoa** [comment]
> scripts into JsonWireprotocol format

Do you expect a particular method call?

**Reply by patchthecode** [reply]
Ok forgive me for my ignorance on the use of this library.
But here is my limited understanding.

1. User writes appium tests in Java language
2. this library converts the test from Java into either (1) jsonWireProtol format or (2) web driver format  (i am not sure)
3. The format (or "file") is sent to a server for processing

This is my limited understanding on how it works. My employer who also has limited knowledge of how this works, has ask me to find out where in this library this conversion from a Java test script into the protocol is taking place.

> Do you expect a particular method call?

Yes.. i was expecting a method call. But i could be wrong since i do not know how this works

**Reply by KazuCocoa** [reply]
Then, I guess some links below may help to track the logic. The base is Appium Java client inherits the selenium client.  It sends MJSONWP/W3C webdriver protocol to the server/driver.

https://github.com/SeleniumHQ/selenium
https://github.com/appium/java-client/blob/master/src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java#L177
https://github.com/appium/java-client/blob/master/src/main/java/io/appium/java_client/MobileCommand.java

**Reply by patchthecode** [reply]
Thanks so much for the response. We will take some time to go through this code/material.

**Reply by patchthecode** [reply]
Hey, we're delving into this today.
just to clarify, 

This repository (Appium Java client) inherits the selenium client right?
And, this Java client, then sends the webdriver protocol over to a server?

My understand from what you have said, is that this client converts a Java appium script file _into_ the webdriver protocol format from this line of code? 
https://github.com/appium/java-client/blob/master/src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java#L177

I guess i am trying to find where the actual javafile written by a java appium tester is converted to be used at line 177?

**Reply by patchthecode** [reply]
I'll create a simple java appium test script, and try to run it with this library to manually step through the code flow.

**Reply by KazuCocoa** [reply]
_converts a Java appium script file into the webdriver protocol format_
I don't understand this meaning correctly... What is the intention of "a Java appium script file"?

A user writes a command like find element:
```
driver.findElement(By.id("IntegerA"));
```
Then, this java client/selenium java client sends this command to an appium server as:
```
HTTP method: POST
URL: https://server:port/wd/hub/session/<session id>/element
body: {"using": "id", "value": "IntegerA"}
```

Then, the appium server replies back the response to the client. The L177's execute command does the sending HTTP request and receive the request.

**Reply by patchthecode** [reply]
thanks.
for this information we found the needed information
